db/migrate/20240305213939_add_prioritized_client_columns_to_match_route_table.rb

```diff
@@ -0,0 +1,30 @@
+class AddPrioritizedClientColumnsToMatchRouteTable < ActiveRecord::Migration[6.1]
+  def change
+    add_column :match_routes, :prioritized_client_columns, :text
+
+      MatchRoutes::Base.
+  joins(:match_prioritization).
+  pluck(:type, 'match_prioritizations.type').
+  map do |route, column|
+    puts route
+    column_list = []
+    priortitization = column.constantize
+    column_list << priortitization.client_prioritization_summary_method if priortitization.client_prioritization_summary_method.present?
+    if priortitization.supporting_data_columns.present?
+      priortitization.supporting_data_columns.values.each do |v|
+        # in this case, the columns are inside a Proc. Read the code to pull the column names
+        file_name = v.source_location[0]
+        line_number = v.source_location[1]
+        code = IO.readlines(file_name)[line_number - 1].strip
+        # Code is in the format: "'Match Group' => ->(client) { client.match_group },"
+        code = code[code.index('{')..code.index('}')]
+        code.slice!('{ client.')
+        code.slice!('?')
+        code.slice!(' }')
+        column_list << code
+      end
+    end
+    route.constantize.update_all(prioritized_client_columns: column_list)
+  end
```

Comment:
This feels a bit too clever (and probably brittle).  
There are only 4 versions of this.  Can we add a method to the schemes where we have extra columns, something like: 

in: MatchPrioritization::DaysHomelessPopulations
```
def self.supporting_column_names
  [
    :family_member,
    :veteran,
    :calculated_is_currently_youth,
    :days_homeless,
  ]
end
```

There's some complexity around the youth one, so that' would probably warrant a method on Client
```
def calculated_is_currently_youth
  is_currently_youth || age&.between?(18, 24) || false
end
```

Then, I think this migration can be something like:
```suggestion
      MatchRoutes::Base.preload(:match_prioritization).each do |route|
    route.update(prioritized_client_columns: route.match_prioritization.supporting_column_names)
  end
```

That's totally untested, so may be way off.